src/main/kotlin/io/github/moulberry/notenoughupdates/miscgui/customtodos/CustomTodoEditor.kt

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2023 NotEnoughUpdates contributors
+ *
+ * This file is part of NotEnoughUpdates.
+ *
+ * NotEnoughUpdates is free software: you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * NotEnoughUpdates is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with NotEnoughUpdates. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.moulberry.notenoughupdates.miscgui.customtodos
+
+import io.github.moulberry.moulconfig.common.IItemStack
+import io.github.moulberry.moulconfig.forge.ForgeItemStack
+import io.github.moulberry.moulconfig.internal.ClipboardUtils
+import io.github.moulberry.moulconfig.observer.ObservableList
+import io.github.moulberry.moulconfig.xml.Bind
+import io.github.moulberry.moulconfig.xml.XMLUniverse
+import io.github.moulberry.notenoughupdates.util.SBInfo
+import io.github.moulberry.notenoughupdates.util.Utils
+import io.github.moulberry.notenoughupdates.util.loadResourceLocation
+import net.minecraft.client.Minecraft
+import net.minecraft.init.Items
+import net.minecraft.item.Item
+import net.minecraft.item.ItemStack
+import net.minecraft.util.ResourceLocation
+
+class CustomTodoEditor(
+    val from: CustomTodo,
+    val todos: ObservableList<CustomTodoEditor>,
+    val xmlUniverse: XMLUniverse
+) {
+    @field:Bind
+    var label: String = from.label
+
+    @field:Bind
+    var enabled: Boolean = from.isEnabledOnCurrentProfile
+
+    @field:Bind
+    var timer: String = from.timer.toString()
+
+    @field:Bind
+    var trigger: String = from.trigger
+
+    @field:Bind
+    var icon: String = from.icon
+
+    @field:Bind
+    var isResetOffset: Boolean = from.isResetOffset
+
+    var target = from.triggerTarget
+    var matchMode = from.triggerMatcher
+
+
+    fun into(): CustomTodo {
+        return CustomTodo(
+            label,
+            timer.toIntOrNull() ?: 0,
+            trigger,
+            icon,
+            isResetOffset,
+            target, matchMode,
+            from.readyAt,
+            from.enabled.toMutableMap().also { it[SBInfo.getInstance().currentProfile] = enabled }
```

Comment:
This crashes when saving with the currentProfile being null (which happens when you open /neutodos and save before joining skyblock)

```
java.lang.NullPointerException: getInstance().currentProfile must not be null
at io.github.moulberry.notenoughupdates.miscgui.customtodos.CustomTodoEditor.into(CustomTodoEditor.kt:73)
at io.github.moulberry.notenoughupdates.miscgui.customtodos.CustomTodoList.save(CustomTodoList.kt:78)
```